cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -75,10 +78,49 @@ func New(endpoints []string, ca, cert, key string) (*Client, error) {
 	return &client, nil
 }
 
+// IsEtcdLinsteningOnLocalHostOnly return true if the etcd manifest have etcd listening on localhost only.
+// Listening on local host only was the default in kubeadm <= v1.12, while starting from v1.13 etcd is listening
+// on localhost and API server advertise address (thus allowing add new member when doing join --control-plane).
+func IsEtcdLinsteningOnLocalHostOnly() bool {
+	etcdManifestFile := constants.GetStaticPodFilepath(constants.Etcd, constants.GetStaticPodDirectory())
+	if _, err := os.Stat(etcdManifestFile); err == nil {
+		klog.V(1).Infoln("checking etcd manifest")
+		etcdPod, err := staticpod.ReadStaticPodFromDisk(etcdManifestFile)
+		if err == nil {
```

Comment:
do we want to handle the error case: `err != nil`?
i guess this will make the function return (error, bool).

Reply:
this bit of code was the same already in place for v1.13, and it never gave problems.
let's keep it as it is because if the etcd manifest is invalid for any reason, the upgrade procedure will fail in many other ways before arriving here